### PR TITLE
Feat(eos_validate_state): ANTA Add eAPI HTTPS SSL profile test

### LINKED ISSUE #2850: Feature: Validate SSL profiles that use certificates
### Enhancement summary

At the moment it is possible to specify certificates to be used in SSL profiles on EOS without these files being present on the device. This matches the behaviour on EOS devices, so it's not an issue with the configuration. However it could be helpful to have the `eos_validate_state` check SSL and Management profiles to ensure that any references to certificates work. 

Additionally it may be helpful to check the expiry data on any certificates in use and advise/alert if it's within a certain amount of days from the present (example 30 days).

### Which component of AVD is impacted

eos_validate_state

### Use case example

Run `eos_validate_state` and get warnings/errors on : 
- missing/mismatching certificate names in SSL and Management profiles
- expired or expiring certificates

### Describe the solution you would like

Refer to use case example

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
## Change Summary

Added ANTA test case to verify eAPI HTTPS SSL profile is configured and state is valid.

## Related Issue(s)

Fixes #2850 

## Component(s) name

`arista.avd.eos_validate_state` ANTA Preview

## Proposed changes
1. Added ANTA test case for eAPI HTTPS SSL profile status

## How to test
Add below details in input group vars to generate the config
```
custom_structured_configuration_management_api_http:
  https_ssl_profile: ARISTA-AVD
```
Once the config pushed on the device then run eos_validate_state molecule.

1. Testcase get pass when a ssl profile is configured with state as valid.
```
test_id,node,test_category,test_description,test,result,failure_reason
1,s1-leaf1,SSL,eAPI HTTPS server SSL profile is configured and valid,SSL,PASS,
2,s1-leaf2,SSL,eAPI HTTPS server SSL profile is configured and valid,SSL,PASS,
3,s1-leaf3,SSL,eAPI HTTPS server SSL profile is configured and valid,SSL,PASS,
4,s1-leaf4,SSL,eAPI HTTPS server SSL profile is configured and valid,SSL,PASS,
5,s1-spine1,SSL,eAPI HTTPS server SSL profile is configured and valid,SSL,PASS,
6,s1-spine2,SSL,eAPI HTTPS server SSL profile is configured and valid,SSL,PASS,
```
2. Testcase get failed with correct message if input profile is not matched or not configured.
```
test_id,node,test_category,test_description,test,result,failure_reason
1,s1-leaf1,SSL,eAPI HTTPS server SSL profile is configured and valid,SSL,FAIL,eAPI HTTPS server SSL profile (Invalid-ARISTA-AVD) is misconfigured or invalid
2,s1-leaf2,SSL,eAPI HTTPS server SSL profile is configured and valid,SSL,FAIL,eAPI HTTPS server SSL profile (Invalid-ARISTA-AVD) is misconfigured or invalid
3,s1-leaf3,SSL,eAPI HTTPS server SSL profile is configured and valid,SSL,FAIL,eAPI HTTPS server SSL profile (Invalid-ARISTA-AVD) is misconfigured or invalid
4,s1-leaf4,SSL,eAPI HTTPS server SSL profile is configured and valid,SSL,FAIL,eAPI HTTPS server SSL profile (Invalid-ARISTA-AVD) is misconfigured or invalid
5,s1-spine1,SSL,eAPI HTTPS server SSL profile is configured and valid,SSL,FAIL,eAPI HTTPS server SSL profile (Invalid-ARISTA-AVD) is not configured
6,s1-spine2,SSL,eAPI HTTPS server SSL profile is configured and valid,SSL,FAIL,eAPI HTTPS server SSL profile (Invalid-ARISTA-AVD) is misconfigured or invalid
```
3. Testcase will not run if ssl profile var is not defined in input group vars.
4. Ansible task get fail in case the SSL profile is invalid as https server not running with invalid profile hence eapi connection broke. Need to look more on it

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestsecurity.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import get
+
+
+class AvdTestHttpsSSL(AvdTestBase):
+    """
+    AvdTestHttpsSSL class for HTTPS SSL tests.
+    """
+
+    anta_module = "anta.tests.security"
+    categories = ["ssl"]
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestsecurity.py]: maybe let's use `security` here?

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestsecurity.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    @cached_property
+    def test_definition(self) -> dict:
+        """
+        Generates the proper ANTA test definition for all SSL tests.
+
+        Returns:
+            test_definition (dict): ANTA test definition.
+        """
+        anta_tests = []
+        profile = get(self.hostvars[self.device_name], "custom_structured_configuration_management_api_http.https_ssl_profile")
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestsecurity.py]: you should not be reading in `custom_structured_configuration` as this is not part of the rendered structured config
```suggestion
        profile = get(self.hostvars[self.device_name], "management_api_http.https_ssl_profile")
```

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestsecurity.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        if profile:
+            anta_tests = [
+                {
+                    "VerifyAPIHttpsSSL": {
+                        "result_overwrite": {"categories": self.categories, "description": self.description, "custom_field": "SSL"},
+                        "profile": profile,
+                    }
+                },
+            ]
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestsecurity.py]:
```suggestion
            anta_tests.append(
                {
                    "VerifyAPIHttpsSSL": {
                        "result_overwrite": {"categories": self.categories, "description": self.description, "custom_field": "SSL"},
                        "profile": profile,
                    }
                }
            )
```

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/constants.py [hunk 4]
```diff
@@ -34,6 +35,7 @@
     AvdTestRoutingTable: {"legacy_ansible_tags": ["routing_table"]},
     AvdTestInbandReachability: {"legacy_ansible_tags": ["loopback_reachability", "loopback0_reachability", "optional"]},
     AvdTestLoopback0Reachability: {"legacy_ansible_tags": ["loopback_reachability", "loopback0_reachability"]},
+    AvdTestHttpsSSL: {"legacy_ansible_tags": ["ssl"]},
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/constants.py]: There was no legacy tags for this so you can leave it empty
```suggestion
    AvdTestHttpsSSL: {},
```

I would prefer if the above works but check that we are not assuming the key always exist ;)

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md
Comment (carl-baillargeon) [inline ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md:1]: Let's add a section new features and include the new tests we'll add.

File: ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md [hunk 5]
```diff
@@ -117,6 +117,9 @@ title: Ansible Collection Role eos_valudate_state - Preview Integration with ANT
 - AvdTestReloadCause (Ansible tags: `reload_cause`, `optional`, `never`)
   - VerifyReloadCause: Validate last reload cause. (Optional)
 
+- AvdTestHttpsSSL (Ansible tags: `ssl`, `profile name`)
```
Comment (carl-baillargeon) [inline ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md]:
```suggestion
- (New) AvdTestHttpsSSL (No Ansible tags: Use the new `skipped_tests` variable instead)
```

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestsecurity.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        Returns:
+            test_definition (dict): ANTA test definition.
+        """
+        anta_tests = []
+        profile = get(self.hostvars[self.device_name], "custom_structured_configuration_management_api_http.https_ssl_profile")
+        if profile:
+            anta_tests = [
+                {
+                    "VerifyAPIHttpsSSL": {
+                        "result_overwrite": {"categories": self.categories, "description": self.description, "custom_field": "SSL"},
```
Comment (carl-baillargeon) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestsecurity.py]:
```suggestion
```
We overwrite the ANTA result only if needed. For example, if we run the same test multiple times with different inputs, we need the `custom_field` to differentiate the tests in the AVD report. If we don't provide `result_overwrite`, the AVD report will keep the native ANTA test `categories` and `description` which is what we want most of the time.

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestsecurity.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+
+class AvdTestHttpsSSL(AvdTestBase):
+    """
+    AvdTestHttpsSSL class for HTTPS SSL tests.
+    """
+
+    anta_module = "anta.tests.security"
+    categories = ["ssl"]
+    description = "eAPI HTTPS server SSL profile is configured and valid"
```
Comment (carl-baillargeon) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestsecurity.py]: These vars are not required (see my comment below about `result_overwrite`).

We use `categories` and `description` from ANTA directly.

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestsecurity.py (hunk 2 above)
Comment (carl-baillargeon) [reply ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestsecurity.py]: Please use the helper methods of `AvdTestBase` to add logging. You could do something like:
```
if (profile := self.logged_get(key="management_api_http.https_ssl_profile", logging_level="INFO") is None:
    return None
``` 
Check `AvdTestMLAG` for reference.

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/constants.py (hunk 4 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/roles/eos_validate_state/python_modules/constants.py]: Empty dict is not working, gives key error for legacy_ansible_tags so updated the get_anta_results file for the same

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/constants.py [hunk 8]
```diff
@@ -17,7 +18,7 @@
     AvdTestRoutingTable,
 )
 
-ACRONYM_CATEGORIES: set[str] = {"aaa", "mlag", "snmp", "bgp", "ospf", "vxlan", "stp", "igmp", "ip", "lldp", "ntp"}
+ACRONYM_CATEGORIES: set[str] = {"aaa", "mlag", "snmp", "bgp", "ospf", "vxlan", "stp", "igmp", "ip", "lldp", "ntp", "security"}
```
Comment (carl-baillargeon) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/constants.py]: Is `security` still needed?

Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/roles/eos_validate_state/python_modules/constants.py]: Removed it, Thanks

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestsecurity.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+class AvdTestAPIHttpsSSL(AvdTestBase):
+    """
+    AvdTestAPIHttpsSSL class for verifies if eAPI HTTPS server SSL profile is configured and valid.
+    """
+
+    anta_module = "anta.tests.security"
+
+    @cached_property
+    def test_definition(self) -> dict:
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestsecurity.py]: Given that we can return None (line 28)

```suggestion
    def test_definition(self) -> dict | None:
```

File: ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md [hunk 10]
```diff
@@ -117,6 +122,9 @@ title: Ansible Collection Role eos_valudate_state - Preview Integration with ANT
 - AvdTestReloadCause (Ansible tags: `reload_cause`, `optional`, `never`)
   - VerifyReloadCause: Validate last reload cause. (Optional)
 
+- (New) AvdTestAPIHttpsSSL (No Ansible tags: Use the new `skipped_tests` variable instead)
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md]:
```suggestion
- (New) AvdTestAPIHttpsSSL (No Ansible tags, use the new `skipped_tests` variable instead)
```

File: ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md [hunk 11]
```diff
@@ -42,6 +42,11 @@ title: Ansible Collection Role eos_valudate_state - Preview Integration with ANT
 !!! note
     Starting from version 4.30.1F, `service_routing_protocols_model` is preset to `multi-agent` by default on EOS devices.
 
+## New Fetures and tests
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md]:
```suggestion
## New features and tests
```

File: ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md [hunk 12]
```diff
@@ -42,6 +42,11 @@ title: Ansible Collection Role eos_valudate_state - Preview Integration with ANT
...
+
+- AvdTestAPIHttpsSSL (No Ansible tags: Use the new `skipped_tests` variable instead)
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md]: So here it looks like we will end up maintaining new tests in two places.
Can we avoid this for maintainability simplicity?

Maybe keep the test in the test categories and indicate in this paragraph that the New tests and categories are marked with the `(New)` string in the section?

Thoughts @carl-baillargeon 

```suggestion
- AvdTestAPIHttpsSSL (No Ansible tags, use the new `skipped_tests` variable instead)
```

Comment (carl-baillargeon) [reply ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md]: Yes I agree with keeping the test in test categories and create a note for the section.

I will add a "New features" only later on once we add new stuff.

Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md]: As suggested removed New features section and added note in test category section

File: ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md [hunk 13]
```diff
@@ -75,6 +75,8 @@ title: Ansible Collection Role eos_valudate_state - Preview Integration with ANT
 
 ## Test Categories
 
+**Note:** New tests are marked with the (New) string.
```
Comment (carl-baillargeon) [inline ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md]: Please use `!!! note` for a nice box in MkDocs.

**Review by carl-baillargeon — APPROVED** [review]
Approved after changing the note in the documentation.

**Review by gmuloc — APPROVED** [review]

File: ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md [hunk 14]
```diff
@@ -75,6 +75,8 @@ title: Ansible Collection Role eos_valudate_state - Preview Integration with ANT
 
 ## Test Categories
 
+!!! Note: New tests are marked with the (New) string.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md]:
```suggestion
!!! note
    New tests are marked with the (New) string.
```

**Review by gmuloc — APPROVED** [review]
Ok if doc is all right when CI finishes

**Pull request merged**